fiftyone/utils/eval/openimages.py

```diff
@@ -0,0 +1,699 @@
+| `voxel51.com <https://voxel51.com/>`_
+|
+"""
+from collections import defaultdict
+import copy
+import logging
+
+import matplotlib.pyplot as plt
+import numpy as np
+import sklearn.metrics as skm
+
+import fiftyone.core.utils as fou
+
+from .detection import (
+    DetectionEvaluation,
+    DetectionEvaluationConfig,
+    DetectionResults,
+)
+
+
+logger = logging.getLogger(__name__)
+
+
+class OpenImagesEvaluationConfig(DetectionEvaluationConfig):
+    """Open Images-style evaluation config.
+
+    Args:
+        pred_field: the name of the field containing the predicted
+            :class:`fiftyone.core.labels.Detections` instances
+        gt_field: the name of the field containing the ground truth
+            :class:`fiftyone.core.labels.Detections` instances
+        iou (None): the IoU threshold to use to determine matches
+        classwise (None): whether to only match objects with the same class
+            label (True) or allow matches between classes (False)
+        iscrowd ("IsGroupOf"): the name of the crowd attribute
+        max_preds (None): the maximum number of predicted objects to evaluate
+            when computing mAP and PR curves. 
+        hierarchy (None): a dict containing a hierachy of classes for
+            evaluation following the structure 
+            ``{"LabelName": label, "Subcategory": [{...}, ...]}``
+        pos_lab_field (None): the name of the field containing 
+            image-level classifications that specify which classes should be 
+            evaluated in the image
+        neg_lab_field (None): the name of the field containing 
+            image-level classifications that specify which classes should not 
+            be evaluated in the image
+        expand_hierarchy (True): bool indicating whether to expand ground truth
+            detections and labels according to the provided hierarchy
+        expand_pred_hierarchy (False): bool indicating whether to expand
+            predicted detections and labels according to the provided 
+            hierarchy
+    """
+
+    def __init__(
+        self,
+        pred_field,
+        gt_field,
+        iou=None,
+        classwise=None,
+        iscrowd="IsGroupOf",
+        max_preds=None,
+        hierarchy=None,
+        pos_lab_field=None,
+        neg_lab_field=None,
+        expand_hierarchy=True,
+        expand_pred_hierarchy=False,
+        **kwargs
+    ):
+        super().__init__(
+            pred_field, gt_field, iou=iou, classwise=classwise, **kwargs
+        )
+
+        if max_preds is None:
+            max_preds = 100
+
+        self.iscrowd = iscrowd
+        self.max_preds = max_preds
+        self.hierarchy = hierarchy
+        self.pos_lab_field = pos_lab_field
+        self.neg_lab_field = neg_lab_field
+        self.expand_hierarchy = expand_hierarchy
+        self.expand_pred_hierarchy = expand_pred_hierarchy
+
+        if expand_pred_hierarchy:
+            if not hierarchy:
+                logger.info(
```

Comment:
suggestion: use `logger.warning()` when you are overriding the user's parameter value